x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/TokenService.java

```diff
@@ -1798,14 +1813,14 @@ static BytesReference createTokenDocument(
                     .startObject("client")
                     .field("type", "unassociated_client")
                     .field("user", originatingClientAuth.getUser().principal())
-                    .field("realm", originatingClientAuth.getAuthenticatedBy().getName())
+                    .field("realm", originatingClientAuth.getSourceRealm().getName())
                     .endObject()
                     .endObject();
             }
             builder.startObject("access_token")
                 .field("invalidated", false)
                 .field("user_token", userToken)
-                .field("realm", userToken.getAuthentication().getAuthenticatedBy().getName())
+                .field("realm", userToken.getAuthentication().getSourceRealm().getName())
```

Comment:
I think this is the change we do want. The token itself represent the effective user.